packages/ai-jsx/src/core/opentelemetry.ts

```diff
@@ -59,8 +63,12 @@ export function openTelemetryStreamRenderer(streamRenderer: StreamRenderer): Str
           throw ex;
         } finally {
           if (result) {
+            const resultIsPartial = result.find(isElement);
             // Record the rendered value.
-            span.setAttribute('ai.jsx.result', result.find(isElement) ? debug(result, true) : result.join(''));
+            span.setAttribute('ai.jsx.result', resultIsPartial ? debug(result, true) : result.join(''));
+            if (!resultIsPartial) {
+              span.setAttribute('ai.jsx.result.tokenCount', getEncoder().encode(result.join('')).length);
```

Comment:
Is there no sensible place to log this from OpenAI or ShrinkConversation? We already log the entire prompt going into the model (I think twice even), and attaching the token counts seems like it would be straightforward.

Seeing a single event with the entire prompt broken down by token count seems easier to grok than needing to inspect N different components.

From a trace perspective if we wanted a span to attach aggregated token counts to, I think the OpenAIChatModel one would make sense to use for both prompt and completion token counts/estimates. (It's straightforward to attach arbitrary attributes to the active span because there's a global way to get the current span.)

Reply:
I agree that we need better visibility into the final prompt. But I also think there's value in getting a token count for each individual component; by the time we get to `OpenAI`, those components may have been rendered away. And `ShrinkConversation` may not even be in use.

So, my suggestion is that we do both this and what you're suggestion.